components/03-modules/modal/modal.js

```diff
@@ -4,45 +4,85 @@ class Modal {
     this.init();
   }
 
-  toggleModal(modalElement) {
-    modalElement.classList.toggle('modal--active')
+  trap(e, modal) {
+    if (e.which == 27) {
+      this.closeModal(modal);
+    }
+    if (e.which == 9) {
+      let currentFocus = document.activeElement,
+          totalOfFocusable = modal.focusableChildren.length,
+          focusedIndex = modal.focusableChildren.indexOf(currentFocus);
+      if (e.shiftKey) {
+        if (focusedIndex === 0) {
+          e.preventDefault();
+          modal.focusableChildren[totalOfFocusable - 1].focus();
+        }
+      }
+      else {
+        if (focusedIndex == totalOfFocusable - 1) {
+          e.preventDefault();
+          modal.focusableChildren[0].focus();
+        }
+      }
+    }
+  }
+
+  openModal(modal) {
+    modal.focused = document.activeElement;
+    modal.el.classList.add(modal.activeClass);
```

Comment:
we have to change `aria-hidden attribute`, please add `modal.el.setAttribute('aria-hidden', false);`

Reply:
done